src/client/sky.h

```diff
@@ -134,25 +155,76 @@ class Sky : public scene::ISceneNode
 	bool m_clouds_visible; // Whether clouds are disabled due to player underground
 	bool m_clouds_enabled = true; // Initialised to true, reset only by set_sky API
 	bool m_directional_colored_fog;
-	bool m_bodies_visible = true; // sun, moon, stars
+	bool m_in_clouds = true; // Prevent duplicating bools to remember old values
+
 	video::SColorf m_bgcolor_bright_f = video::SColorf(1.0f, 1.0f, 1.0f, 1.0f);
 	video::SColorf m_skycolor_bright_f = video::SColorf(1.0f, 1.0f, 1.0f, 1.0f);
 	video::SColorf m_cloudcolor_bright_f = video::SColorf(1.0f, 1.0f, 1.0f, 1.0f);
 	video::SColor m_bgcolor;
 	video::SColor m_skycolor;
 	video::SColorf m_cloudcolor_f;
-	v3f m_stars[SKY_STAR_COUNT];
+	
+	// Horizon colors
+	video::SColorf m_bgcolor_day_f = video::SColor(255, 155, 193, 240);
```

Comment:
I highly recommend separating state from drawing logic. By this, I mean that this should have a separate class called SkyParams which contains all the stuff needed by set_sky, and an object called SkyObjectParams which contains the stuff needed by set_moon and set_sun (one contains the subset of another). This would simplify the gross code in client/event.